### PR TITLE
Translating API tests to Python — part 1

### PR DESCRIPTION
**Pull request by yoni206**
This PR translates some of the API tests from [here](https://github.com/cvc5/cvc5/tree/master/test/api) to Python. The other tests are translated in a [private branch](https://github.com/yoni206/cvc5/tree/python_dir_struct/test/python/api) and will be added in a separate PR.

### CODE REVIEW

File: test/python/CMakeLists.txt [hunk 1]
```diff
@@ -26,12 +26,29 @@ add_test (NAME python/unit/api/${name}
   WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/src/api/python)
```
Comment (aniemetz) [inline test/python/CMakeLists.txt:26]:
```suggestion
  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/src/api/python)
  set_tests_properties(python/unit/api/${name} PROPERTIES LABELS "unit")
```

File: test/python/CMakeLists.txt [hunk 2]
```diff
@@ -26,12 +26,29 @@ add_test (NAME python/unit/api/${name}
...
 
+# Add Python bindings API tests.
+macro(cvc5_add_python_api_test name filename)
+# We create test target 'python/api/myapitest' and run it with
+# 'ctest -R "python/api/myapitest"'.
+add_test (NAME python/api/${name}
+  COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/${filename}
+  # directory for importing the python bindings
+  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/src/api/python)
+  set_tests_properties(python/api/${name} PROPERTIES
```
Comment (aniemetz) [inline test/python/CMakeLists.txt:37]:
```suggestion
  set_tests_properties(python/api/${name} PROPERTIES
      LABELS "api"
```

File: test/api/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ###############################################################################
 # Top contributors (to current version):
-#   Aina Niemetz, Andrew V. Jones, Gereon Kremer
+#   Yoni Zohar
```
Comment (yoni206) [inline test/api/CMakeLists.txt:3]:
```suggestion
#   Aina Niemetz, Andrew V. Jones, Gereon Kremer
```

**Comment by yoni206** [comment]
@aniemetz , I have addressed your review and reorganized the dir/file structure.

File: test/unit/api/python/CMakeLists.txt [hunk 4]
```diff
@@ -16,10 +16,10 @@
 # Check if the pytest Python module is installed.
 check_python_module("pytest")
 
-# Add Python bindings API tests.
+# Add Python bindings API unit tests.
 macro(cvc5_add_python_api_unit_test name filename)
-# We create test target 'python/unit/api/myapitest' and run it with
-# 'ctest -R "python/unit/api/myapitest"'.
+# We create test target 'unit/api/python/myapitest' and run it with
+# 'ctest -R "unit/api/python/myapitest"'.
```
Comment (aniemetz) [inline test/unit/api/python/CMakeLists.txt:22]: Can you add label `python` to line 29? Like this: "unit python"

**Review by aniemetz — COMMENTED** [review]
`interactive_shell.py` should not be in directory `cpp` but `python`

**Comment by nafur** [comment]
> `interactive_shell.py` should not be in directory `cpp` but `python`

Actually, this is a python script but it tests the cvc5 binary. As such, I'd argue it doesn't even belong in the api folder...

**Comment by aniemetz** [comment]
Indeed. Where should we put it?

**Comment by yoni206** [comment]
I thought about creating a folder for it but it seems like an overkill just
for a single test, so I left it with the cpp. But I am happy to move it
wherever you find suitable.

**Comment by aniemetz** [comment]
We decided offline that it would probably best to move it to `test/binary`. @yoni206 can you include this move here?

**Comment by yoni206** [comment]
@aniemetz sure, done [here](ad8d1dbb004ca1188b6f46a1da192657e6ef9b0b).

File: test/binary/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+# #############################################################################
+#
+# The build system configuration.
+##
+include_directories(.)
+include_directories(${PROJECT_SOURCE_DIR}/src)
+include_directories(${PROJECT_SOURCE_DIR}/src/include)
+include_directories(${CMAKE_BINARY_DIR}/src)
+
+add_custom_target(binarytests
```
Comment (aniemetz) [inline test/binary/CMakeLists.txt:20]: This should be guarded (moved below). If no pexpect, the target will not find any test with label `binary`.

Comment (yoni206) [reply test/binary/CMakeLists.txt]: Addressed in a8426434ad163d00f0207da706960459bcfa4738 .

**Review by aniemetz — APPROVED** [review]

**Pull request merged**